Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -314,11 +318,93 @@ void CookieStore::remove(CookieStoreDeleteOptions&& options, Ref<DeferredPromise
     set(WTFMove(initOptions), WTFMove(promise));
 }
 
+void CookieStore::cookiesAdded(const String& host, const Vector<Cookie>& cookies)
+{
+    UNUSED_PARAM(host);
+    if (!m_hasChangeEventListener)
+        return;
+
+    auto* context = scriptExecutionContext();
+    if (!context)
+        return;
+
+    auto& document = *downcast<Document>(context);
+    UNUSED_PARAM(document);
+    ASSERT(host == document.url().host().toString());
+
+    Vector<CookieListItem> changedCookieList;
+    Vector<CookieListItem> deletedCookieList;
+    auto currentTime = WallTime::now().secondsSinceEpoch().milliseconds();
+    for (auto cookie : cookies) {
```

Comment:
auto&

Reply:
The compiler doesn't let me do this. It complains because with `auto`, the it treats the cookie as const and won't allow the below line `cookie.value = nullString();` to work. I have left it as just `auto` for now. Is this ok?

Reply:
Ok